workflow/artifacts/oss/oss.go

```diff
@@ -315,33 +316,32 @@ func setBucketLifecycleRule(client *oss.Client, ossArtifact *wfv1.OSSArtifact) e
 		return fmt.Errorf("markInfrequentAccessAfterDays cannot be large than markDeletionAfterDays")
 	}
 
-	// Set expiration rule.
-	expirationRule := oss.BuildLifecycleRuleByDays("expiration-rule", ossArtifact.Key, true, markInfrequentAccessAfterDays)
-	// Automatically delete the expired delete tag so we don't have to manage it ourselves.
+	// Delete the current version objects after a period of time.
+	// If BucketVersioning is enbaled, the objects will turn to non-current version.
 	expiration := oss.LifecycleExpiration{
-		ExpiredObjectDeleteMarker: pointer.Bool(true),
+		Days: markDeletionAfterDays,
 	}
 	// Convert to Infrequent Access (IA) storage type for objects that are expired after a period of time.
-	versionTransition := oss.LifecycleVersionTransition{
-		NoncurrentDays: markInfrequentAccessAfterDays,
-		StorageClass:   oss.StorageIA,
+	transition := oss.LifecycleTransition{
+		Days:         markInfrequentAccessAfterDays,
+		StorageClass: oss.StorageIA,
 	}
-	// Mark deletion after a period of time.
-	versionExpiration := oss.LifecycleVersionExpiration{
-		NoncurrentDays: markDeletionAfterDays,
+	// Delete the aborted uploaded parts after a period of time.
+	abortMultipartUpload := oss.LifecycleAbortMultipartUpload{
+		Days: markDeletionAfterDays,
 	}
-	versionTransitionRule := oss.LifecycleRule{
-		ID:                    "version-transition-rule",
-		Prefix:                ossArtifact.Key,
-		Status:                string(oss.VersionEnabled),
-		Expiration:            &expiration,
-		NonVersionExpiration:  &versionExpiration,
-		NonVersionTransitions: []oss.LifecycleVersionTransition{versionTransition},
+
+	rule := oss.LifecycleRule{
+		ID:                   "argo-workflows-rule",
```

Comment:
Does this ID need to correspond to artifacts?

Reply:
OSS ArtifactDriver could only receive the name of key and setting of OSSBucket, so I used sha256(key) as ID which could avoid exceeding length limit.